libbeat/service/service.go

```diff
@@ -72,11 +74,49 @@ func BeforeRun() {
 	if *httpprof != "" {
 		go func() {
 			logp.Info("start pprof endpoint")
-			logp.Info("finished pprof endpoint: %v", http.ListenAndServe(*httpprof, nil))
+			mux := http.NewServeMux()
+
+			// register pprof
+			mux.HandleFunc("/debug/pprof/", pprofNet.Index)
```

Comment:
It's kind of a trade of here. By default httpprof just uses the global default handler. Any package you import might decide to register a handler. E.g. pprof, expvar and go-metrics all registers into the default handler (not sure about other packages). Idea is to take control on what's getting exposed via httpprof (plus I need to overwrite `/debug/vars` to not use `expvar`, but `libbeat/monitoring`).

Unfortunately the `net/http/pprof` package does not make the handlers available via a `http.ServerMux`, but low-level handlers callbacks only :(